apps/chat/src/components/Chat/Publish/PublicationHandler/PublicationHandler.tsx

```diff
@@ -0,0 +1,337 @@
+import {
+  ApplicationPublicationResources,
+  ConversationPublicationResources,
+  FilePublicationResources,
+  PromptPublicationResources,
+} from './ReviewResources';
+
+import isEqual from 'lodash-es/isEqual';
+
+interface Props {
+  publication: Publication;
+}
+
+const sections = [
+  {
+    featureType: FeatureType.Chat,
+    sectionName: translate('Conversations'),
+    dataQa: 'conversations-to-approve',
+    Component: ConversationPublicationResources,
+  },
+  {
+    featureType: FeatureType.Prompt,
+    sectionName: translate('Prompts'),
+    dataQa: 'prompts-to-approve',
+    Component: PromptPublicationResources,
+  },
+  {
+    featureType: FeatureType.Application,
+    sectionName: translate('Applications'),
+    dataQa: 'applications-to-approve',
+    Component: ApplicationPublicationResources,
+  },
+  {
+    featureType: FeatureType.File,
+    sectionName: translate('Files'),
+    dataQa: 'files-to-approve',
+    Component: FilePublicationResources,
+  },
+];
+
+export function PublicationHandler({ publication }: Props) {
+  const dispatch = useAppDispatch();
+
+  const { t } = useTranslation(Translation.Chat);
+
+  const [isCompareModalOpened, setIsCompareModalOpened] = useState(false);
+
+  const rules = useAppSelector((state) =>
+    PublicationSelectors.selectRulesByPath(state, publication.targetFolder),
+  );
+  const isRulesLoading = useAppSelector(
+    PublicationSelectors.selectIsRulesLoading,
+  );
+  const isApplicationReview = useAppSelector(
+    PublicationSelectors.selectIsApplicationReview,
+  );
+  const isPublicationUpdating = useAppSelector(
+    PublicationSelectors.selectIsPublicationUpdating,
+  );
+  const editState = useAppSelector(PublicationSelectors.selectEditState);
+
+  const publicationAuthor = useMemo(() => {
+    return extractNameFromEmail(publication.author) ?? t('Unknown');
+  }, [publication.author, t]);
+
+  useEffect(() => {
+    if (publication.targetFolder !== PUBLIC_URL_PREFIX) {
+      dispatch(
+        PublicationActions.uploadRules({
+          path: publication.targetFolder.split('/').slice(1).join('/'),
+        }),
+      );
+    }
+  }, [dispatch, publication.targetFolder]);
+
+  const filteredRuleEntries = useMemo(() => {
+    const rulesEntries = Object.entries(rules);
+    return !publication.rules
+      ? rulesEntries
+      : rulesEntries.filter(([path]) => path !== publication.targetFolder);
+  }, [publication.rules, rules, publication.targetFolder]);
+
+  const newRules: PublicationRule[] = useMemo(
+    () =>
+      publication.rules?.map((rule) => ({
+        source: rule.source,
+        function: rule.function,
+        targets: rule.targets,
+      })) ?? [],
+    [publication.rules],
+  );
+
+  const handleUpdateRequest = useCallback(() => {
+    dispatch(
+      PublicationActions.updatePublicationRequest({
+        url: publication.url,
+        dataToUpdate: {
+          name: publication.name ?? '',
```

Comment:
I think it would be better to send not empty string.

```suggestion
          name: publication.name ?? `New request by ${publication.author}`,
```